libethash-cl/ethash_cl_miner.cpp

```
@@ -78,16 +78,30 @@ ethash_cl_miner::~ethash_cl_miner()
 	finish();
 }
 
+// Quite ugly. Saves us a lot of typing on these static functions
+// since we can't keep the platforms vector in the class (static class/functions)
+//
+// LTODO: With a bit of general refactoring this could go away and platforms
+// be queried only once and kept in the class.
+#define ETHASHCL_GET_PLATFORMS(platforms_, failStmt_)	\
+	do {												\
+		try												\
+		{												\
+			cl::Platform::get(&platforms_);				\
+		}												\
+		catch (cl::Error const& err)					\
+		{												\
+			int errCode = err.err();					\
+			if (errCode == CL_PLATFORM_NOT_FOUND_KHR)	\
+				ETHCL_LOG("No OpenCL platforms found");	\
+			failStmt_;									\
```

You probably want to rethrow the exception in other cases.